src/meta/src/rpc/service/ddl_service.rs

```diff
@@ -688,45 +697,57 @@ where
-            // construct the rewrite mapping for brokers
-            for (link, broker) in link_info.infos.iter().zip_eq_fast(broker_addrs.into_iter()) {
-                let conn = self
-                    .catalog_manager
-                    .get_connection_by_name(&link.service_name)
-                    .await?;
+            let link_targets: Vec<AwsPrivateLinkItem> =
+                serde_json::from_str(link_target_value).map_err(|e| anyhow!(e))?;
 
-                if let Some(info) = conn.info {
-                    match info {
-                        connection::Info::PrivateLinkService(svc) => {
-                            if svc.dns_entries.is_empty() {
-                                return Err(MetaError::from(anyhow!(
-                                    "No available private link endpoints for Kafka broker {}",
-                                    broker
-                                )));
-                            }
-                            let default_dns = svc.dns_entries.values().next().unwrap();
-                            let target_dns = svc.dns_entries.get(&link.availability_zone);
-                            match target_dns {
-                                None => {
-                                    broker_rewrite_map.insert(
-                                        broker.to_string(),
-                                        format!("{}:{}", default_dns, link.port),
-                                    );
-                                }
-                                Some(dns_name) => {
-                                    broker_rewrite_map.insert(
-                                        broker.to_string(),
-                                        format!("{}:{}", dns_name, link.port),
-                                    );
-                                }
-                            }
-                        }
+            let conn = self
+                .catalog_manager
+                .get_connection_by_name(&private_link_name)
+                .await?;
+
+            if let Some(connection::Info::PrivateLinkService(svc)) = &conn.info {
+                // check whether the VPC endpoint is ready
+                let cli = self.aws_client.as_ref().unwrap();
+                if !cli.is_vpc_endpoint_ready(&svc.endpoint_id).await? {
+                    return Err(MetaError::from(anyhow!(
+                        "Private link endpoint {} is not ready",
+                        svc.endpoint_id
+                    )));
+                }
+            } else {
+                return Err(MetaError::from(anyhow!(
+                    "Connection {} is not a private link service",
+                    private_link_name
+                )));
+            }
+
+            // construct the rewrite mapping for brokers
+            for (link, broker) in link_targets.iter().zip_eq_fast(broker_addrs.into_iter()) {
```

Comment:
Do we check `broker_addrs.len() == link_targets.len()` somewhere and report an error to user? If not, `zip_eq_fast` can easily panic and bring down meta service if user makes a mistake.

Reply:
Fixed. Thanks for pointing out it.